### PR TITLE
Agent: consistent 'containerID' field in the log of the requests EP-delete and EP-create

### LINKED ISSUE #13658: Use consistent structured log during endpoint-delete and endpoint-create
[EP create](https://github.com/cilium/cilium/blob/85dea57961a0b2b7a1291eabc7dc79682c966b79/pkg/endpoint/manager.go#L74):

```
level=info msg="New endpoint" containerID=77ce9db8c9 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=3100 identity=5 ipv4=172.18.12.151 ipv6= k8sPodName=default/pod subsys=endpoint
```

vs 

[EP delete](https://github.com/cilium/cilium/blob/85dea57961a0b2b7a1291eabc7dc79682c966b79/daemon/cmd/endpoint.go#L696):

```
level=info msg="Delete endpoint request" id="container-id:7eab59b2e768e9845af3988be879558186e3215e018577c139a3405cbd424e37" subsys=daemon
```

The container ID field is not consistent which makes searching for corresponding deletes or creates based on container ID needlessly difficult. We should use the same format if the ID is a [container ID (not an IP address)](https://github.com/cilium/cilium/blob/85dea57961a0b2b7a1291eabc7dc79682c966b79/daemon/cmd/endpoint.go#L695).

In general, we should also make sure to either use the long-form or short-form container ID.

### PR DESCRIPTION
**Pull request by romanspb80**
Previously, it was difficult to find creates and deletes based on container ID due to non-consistent container ID field in the log: 'EP create' used short-form container ID, 'EP delete' used standart form. For examle:

'EP create':
level=info msg="New endpoint" containerID=77ce9db8c9 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=3100 identity=5 ipv4=172.18.12.151 ipv6= k8sPodName=default/pod subsys=endpoint

vs

'EP delete':
level=info msg="Delete endpoint request" id="container-id:7eab59b2e768e9845af3988be879558186e3215e018577c139a3405cbd424e37" subsys=daemon

This patch fixes it: 'EP delete' uses short-form container ID as 'EP create'.

Fixes: #13658

Signed-off-by: Roman Ptitcyn <romanspb@yahoo.com>

### CODE REVIEW

File: daemon/cmd/endpoint.go [hunk 1]
```diff
@@ -709,6 +709,14 @@ func (d *Daemon) DeleteEndpoint(id string) (int, error) {
 	} else if err = endpoint.APICanModify(ep); err != nil {
 		return 0, api.Error(DeleteEndpointIDInvalidCode, err)
 	} else {
+		containerID := ep.GetShortContainerID()
+		if containerID != "" {
+			log.WithField("containerID", containerID).Info("Delete endpoint request")
```
Comment (sayboras) [inline daemon/cmd/endpoint.go]: maybe containerID ->  logfields.ContainerID for consistency.

File: daemon/cmd/endpoint.go [hunk 2]
```diff
@@ -709,6 +709,14 @@ func (d *Daemon) DeleteEndpoint(id string) (int, error) {
...
+		} else {
+			// if the containerID does not exist, then the IP address is inserted in the log
+			log.WithField("IPv4", ep.GetIPv4Address()).WithField("IPv6", ep.GetIPv6Address()).Info(
```
Comment (sayboras) [inline daemon/cmd/endpoint.go]: same as above, IPv4 -> logfields.IPv4, IPv6 -> logfields.IPv6.

File: daemon/cmd/endpoint.go [hunk 3]
```diff
@@ -709,6 +709,14 @@ func (d *Daemon) DeleteEndpoint(id string) (int, error) {
...
+			"Delete endpoint request")
```
Comment (sayboras) [inline daemon/cmd/endpoint.go]: gofmt failure here

**Review by sayboras — COMMENTED** [review]
Any reason why the below line is still there ? I am under assumption that this should be removed/updated as mentioned in github issue.

https://github.com/cilium/cilium/blob/4c827bf7d1e2c0f2a5ff4c6b606f7c17bda316fa/daemon/cmd/endpoint.go#L702-L703

**Comment by romanspb80** [comment]
> Any reason why the below line is still there ? I am under assumption that this should be removed/updated as mentioned in github issue.
> 
> https://github.com/cilium/cilium/blob/4c827bf7d1e2c0f2a5ff4c6b606f7c17bda316fa/daemon/cmd/endpoint.go#L702-L703

Thanks Tam!
I'll update this

File: daemon/cmd/endpoint.go (hunk 2 above)
Comment (romanspb80) [reply daemon/cmd/endpoint.go]: This is a good idea. I'll do it.

File: daemon/cmd/endpoint.go (hunk 1 above)
Comment (romanspb80) [reply daemon/cmd/endpoint.go]: OK. Thanks!

File: daemon/cmd/endpoint.go (hunk 3 above)
Comment (romanspb80) [reply daemon/cmd/endpoint.go]: No. It works

File: daemon/cmd/endpoint.go [hunk 4]
```diff
@@ -709,6 +710,19 @@ func (d *Daemon) DeleteEndpoint(id string) (int, error) {
 	} else if err = endpoint.APICanModify(ep); err != nil {
 		return 0, api.Error(DeleteEndpointIDInvalidCode, err)
 	} else {
+		containerID := ep.GetShortContainerID()
+		msg := "Deleting endpoint"
```
Comment (christarazi) [inline daemon/cmd/endpoint.go]: Let's keep the log msg the same as it was: `Delete endpoint request`

File: daemon/cmd/endpoint.go [hunk 5]
```diff
@@ -709,6 +710,19 @@ func (d *Daemon) DeleteEndpoint(id string) (int, error) {
...
+		case containerID == "":
+			log.WithFields(logrus.Fields{
+				logfields.IPv4:   ep.GetIPv4Address(),
+				logfields.IPv6:   ep.GetIPv6Address(),
+			}).Info(msg)
+		default:
+			log.WithFields(logrus.Fields{
+				logfields.ContainerID: containerID,
+			}).Info(msg)
+		}
```
Comment (christarazi) [inline daemon/cmd/endpoint.go]:
```suggestion
		msg := "Deleting endpoint"
		switch containerID := ep.GetShortContainerID(); containerID {
		case "":
		        log.WithFields(logrus.Fields{
		    		logfields.IPv4:   ep.GetIPv4Address(),
				logfields.IPv6:   ep.GetIPv6Address(),
			}).Info(msg)
		default:
			log.WithFields(logrus.Fields{
				logfields.ContainerID: containerID,
			}).Info(msg)
		}
```

**Review by christarazi — CHANGES_REQUESTED** [review]
Thanks for the PR! Change looks good overall, a few minor nits. As Tam pointed out, seems like the linter is failing on `gofmt`. Also, please squash your commits into one.

File: daemon/cmd/endpoint.go (hunk 3 above)
Comment (romanspb80) [reply daemon/cmd/endpoint.go]: > gofmt failure here

Sorry. I misunderstood this issue.

File: daemon/cmd/endpoint.go (hunk 5 above)
Comment (romanspb80) [reply daemon/cmd/endpoint.go]: OK
Done.
Thanks

File: daemon/cmd/endpoint.go [hunk 6]
```diff
@@ -699,8 +699,9 @@ func (d *Daemon) deleteEndpointQuiet(ep *endpoint.Endpoint, conf endpoint.Delete
 }
 
 func (d *Daemon) DeleteEndpoint(id string) (int, error) {
-	// id can be an endpoint iD or an IP so not using logfields.EndpointID
-	log.WithField("id", id).Info("Delete endpoint request")
+	// The log message about request "DELETE /endpoint/{id} request" with level=debug has been
+	// in the previous step (deleteEndpointID Handle).
+	// So now needs a log message (level=info) with the information about EP to be deleted.
```
Comment (christarazi) [inline daemon/cmd/endpoint.go]: I'm not sure I understand the need for these comments?

**Review by christarazi — COMMENTED** [review]
One more minor comment. Also, there are a few typos in the commit msg: `examle` and `standart`. Otherwise, this is good to go for me.

File: daemon/cmd/endpoint.go (hunk 6 above)
Comment (romanspb80) [reply daemon/cmd/endpoint.go]: I think too. I made them as an update.
So I've removed them.

**Review by christarazi — APPROVED** [review]

**Comment by christarazi** [comment]
test-me-please

**Review by sayboras — APPROVED** [review]
LGTM :100:

**Comment by romanspb80** [comment]
> LGTM 💯

Thanks :)

**Review by jrajahalme — APPROVED** [review]

**Comment by jrajahalme** [comment]
retest-gke

**Comment by jrajahalme** [comment]
retest-net-next

**Comment by sayboras** [comment]
Hit https://github.com/cilium/cilium/issues/12511

**Comment by errordeveloper** [comment]
I am not sure if `K8s-1.13-Kernel-netnext ` is a flake, does anyone know?

It fails on this:
```
[Fail] K8sServicesTest Checks service across nodes Tests NodePort BPF Tests with vxlan [It] Tests with secondary NodePort device 
19:42:40  /home/jenkins/workspace/Cilium-PR-K8s-1.13-net-next/src/github.com/cilium/cilium/test/k8sT/Services.go:892
```

**Comment by errordeveloper** [comment]
> Hit #12511

Looks like `K8s-1.13-Kernel-netnext` is indeed hitting this well-known flake.

```
[Fail] K8sServicesTest Checks service across nodes Tests NodePort BPF Tests with vxlan [It] Tests with secondary NodePort device 
19:42:40  /home/jenkins/workspace/Cilium-PR-K8s-1.13-net-next/src/github.com/cilium/cilium/test/k8sT/Services.go:892
```

Based 2 reviews, this is good to merge, I believe :) Adding ready-to-merge label for @cilium/janitors to pick this up.

**Review by kaworu — APPROVED** [review]

**Pull request merged**